docs/adr/0008-use-tracing.md

```diff
@@ -0,0 +1,165 @@
+            .with(fmt::layer())
+            .with(EnvFilter::from_default_env())
+            .init();
+```
+
+(probably via a helper of some sort)
+
+### Implement a `tracing_subscriber::Layer`-based "subscriber" mechanism.
+
+This subscriber mechanism would require the application to know all tracing `targets` it cares about - and by default,
+each crate is its own target. The app must explicitly "subscribe" to all such targets individually.
+It will *not* be possible to subscribe to all events for performance reasons, and it's unlikely we'll allow
+any "matching" capabilities (eg, regular expressions or similar) - each target will use an exact string match.
+
+When subscribing to a target, the app will also need to specify the max-level for that target. In other words, a
+subscription can be considered a tuple of `(target, level)`.
+
+This implies that checking whether an individual tracing event has a subscriber will be a `HashMap` lookup for the target name,
+then a numeric comparison for the level. Each target will allow only one subscriber, identified by the `target`.
+This means that changing the level for an existing subscriber will involve replacing the subscription with a new one which holds the new level.
+It is expected that the implementation will simply be "set a new subscription", which will replace the old subscription.
+
+This assumes that the application will subscribe with the actual level needed by that subscriber - ie, that apps will not
+simply subscribe to the highest log level and perform additional filtering after the log event has been dispatched.
+This means that the "performance critical path" for this process is determining if we have a subscriber;
+once we have determined a subscriber matches an event, we can perform relatively expensive operations
+on the event because we assume action will be taken on the event.
+
+An example of these "relatively expensive" operations is fetching event "fields", such as the message or other meta-data.
+
+You will note that this is generic enough to handle traditional "log" messages, our error-reporting logs, and more structured logging.
+The shape of the structs etc (eg, try and use one struct for all use-cases, one struct per use-case, etc) is intentionally omitted to avoid bike-shedding on those details in the first instance.
+
+### Replace all existing "subscribers"
+
+There are 3 main places which would change in the first instance, broken down by platform
+
+#### Mobile
+
+The existing [rust_log_forwarder](https://github.com/mozilla/application-services/tree/main/components/support/rust-log-forwarder) and [error-support](https://github.com/mozilla/application-services/tree/main/components/support/error) crates would change how they collect "events" to using `tracing-subscriber`. Initially we'd not change the API exposed to the mobile apps, but very soon after declaring success on Desktop we'd look to consolidate the APIs as seen by the applications.
```

Comment:
This is really nice.  Maybe it could be called out even more by just saying "Initially, there will be no changes needed."  Then we could point the user towards the Application Services section for the internal changes.